site/src/components/MDXPageContent.tsx

```diff
@@ -1,30 +1,138 @@
-import { VFC } from "react";
+import { useEffect, useState, VFC } from "react";
+import { unstable_batchedUpdates } from "react-dom";
+import { useRouter } from "next/router";
 import { MDXRemote, MDXRemoteSerializeResult } from "next-mdx-remote";
 import Box, { BoxProps } from "@mui/material/Box";
 import { mdxComponents } from "../util/mdxComponents";
 import { INFO_CARD_WIDTH } from "./InfoCard/InfoCardWrapper";
+import MDXPageContext, { Heading } from "./context/MDXPageContext";
 
 type MDXPageContentProps = {
   serializedPage: MDXRemoteSerializeResult<Record<string, unknown>>;
 } & BoxProps;
 
+let detectHeadingFromScrollTimer: NodeJS.Timeout | undefined = undefined;
+
 export const MDXPageContent: VFC<MDXPageContentProps> = ({
   serializedPage,
   ...boxProps
-}) => (
-  <Box
-    {...boxProps}
-    sx={{
-      width: "100%",
-      overflow: "scroll",
-      "& > :not(.info-card-wrapper), > a:not(.info-card-wrapper) > *": {
-        maxWidth: {
-          xs: "100%",
-          sm: `calc(100% - ${INFO_CARD_WIDTH}px)`,
-        },
-      },
-    }}
-  >
-    <MDXRemote {...serializedPage} components={mdxComponents} />
-  </Box>
-);
+}) => {
+  const router = useRouter();
+
+  const [headings, setHeadings] = useState<Heading[]>([]);
+
+  useEffect(() => {
+    setHeadings([]);
+  }, [serializedPage]);
```

Comment:
I wonder if we can avoid this if `useMDXHeading` can both register and unregister a heading. But this is probably not worth a change at this point.